snuba/state.py

```diff
@@ -272,11 +275,21 @@ def get_config_changes():
 
 # Query Recording
 
+
+def safe_dumps_default(value: Any) -> Any:
+    if isinstance(value, Mapping):
+        return {**value}
+    raise TypeError(f'Cannot convert object of type {type(value).__name__} to JSON-safe type')
+
+
+safe_dumps = partial(json.dumps, for_json=True, default=safe_dumps_default)
```

Comment:
I also removed the `default` here just to be extra sure the test would fail without it, and it did.